scripts/pre_commit_linter.py

```diff
@@ -342,10 +361,57 @@ def _pre_commit_linter():
     summary_messages.append(py_result.get())
     print '\n'.join(summary_messages)
     print ''
+    return summary_messages
+
+
+# This part is used for detecting bad patterns.
+def _check_bad_patterns(all_files, summary_messages):
+    print 'Starting Pattern Checks'
+    print '----------------------------------------'
+    total_files_checked = 0
+    total_error_count = 0
+    all_files = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)]
+    for filename in all_files:
+        with open(filename) as f:
+            content = f.read()
+            total_files_checked += 1
+            failed = False
+            for pattern in BAD_PATTERNS:
+                if pattern in content:
+                    failed = True
+                    print '%s --> %s' % (
+                        filename, BAD_PATTERNS[pattern])
+                    total_error_count += 1
+            if failed:
```

Comment:
Same issue. Aren't you executing this once for each file? There'll be a lot of logs in the terminal. I would, however, expect just a single summary message for this entire function.